File: Algorithms/Python3.x/26-Remove_Duplicates_from_Sorted_Array.py
# Runtime: 52 ms, faster than 98.36% of Python3 online submissions for Remove Duplicates from Sorted Array.
# Memory Usage: 14.8 MB, less than 45.41% of Python3 online submissions for Remove Duplicates from Sorted Array.
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0: return 0
        i = 0
        for num in nums[1:]:
            if num != nums[i]: nums[i + 1], i = num, i + 1
        return i + 1


# 双指针

File: Algorithms/C++/26-Remove_Duplicates_from_Sorted_Array.cpp
// Runtime: 20 ms, faster than 93.27% of C++ online submissions for Remove Duplicates from Sorted Array.
// Memory Usage: 9.8 MB, less than 97.50% of C++ online submissions for Remove Duplicates from Sorted Array.
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 0)
            return 0;
        int i = 0;
        for (int j = 1; j < nums.size(); ++j)
            if (nums[i] != nums[j]) {
                ++i;
                nums[i] = nums[j];
            }
        return i + 1;
    }
};
